doc/customizing-install.md

```diff
@@ -0,0 +1,85 @@
+# Customizing installation
+
+## Hooking coreos-installer at boot time
+
+When coreos-installer is run from a CoreOS live image (ISO or PXE) using
+kernel command-line arguments, additional custom code can be run before or
+after the installer.  To do this, specify an Ignition config to the live
+boot that runs the installer.  This config is separate and distinct from the
+Ignition config that governs the installed system.
+
+This is a sample Fedora CoreOS Config with hooks that run both before and
+after the installer:
+
+```
+variant: fcos
+version: 1.1.0
+
+storage:
+  files:
+    - path: /usr/local/bin/pre-install-hook
+      mode: 0755
+      contents:
+        inline: |
+          #!/bin/bash
+
+          sleep 10
+          echo "pre-hook"
+    - path: /usr/local/bin/post-install-hook
+      mode: 0755
+      contents:
+        inline: |
+          #!/bin/bash
+
+          sleep 10
+          echo "post-hook"
+
+systemd:
+  units:
+    - name: pre-install-hook.service
+      enabled: true
+      contents: |
+        [Unit]
+        Description=Run before install
+        Before=coreos-installer.service
```

Comment:
To make the two hooks symmetrical, WDYT about:

1. using `RequiredBy=coreos-installer.target` on both units
2. using `Before=coreos-installer.target` on both units (I think this is actually implied by `RequiredBy=` for target units BTW, but it doesn't hurt either -- see https://www.freedesktop.org/software/systemd/man/systemd.target.html#Default%20Dependencies).

So essentially the only difference is the `Before=` and `After=` for `coreos-installer.service`.

Reply:
Also, should we recommend `After=coreos-installer-pre.target` in both?

Reply:
`coreos-installer-pre.target` runs so early that we shouldn't need to sequence on it.  The pre-hook needs `RequiredBy=coreos-installer.service` because we shouldn't run the install if a pre-hook fails.

Reply:
Actually, the single unit currently pulled in by `coreos-installer-pre.target` runs early, but the target is not constrained to do so in future.  I've added `After` to the pre-hook; it's redundant for the post-hook.